pkg/controller/elasticsearch/services/services_test.go

```diff
@@ -152,3 +156,103 @@ func TestElasticsearchURL(t *testing.T) {
 		})
 	}
 }
+
+func TestNewExternalService(t *testing.T) {
+	testCases := []struct {
+		name     string
+		httpConf commonv1beta1.HTTPConfig
+		wantSvc  func() corev1.Service
+	}{
+		{
+			name: "no TLS",
+			httpConf: commonv1beta1.HTTPConfig{
+				TLS: commonv1beta1.TLSOptions{
+					SelfSignedCertificate: &commonv1beta1.SelfSignedCertificate{
+						Disabled: true,
+					},
+				},
+			},
+			wantSvc: mkService,
+		},
+		{
+			name: "self-signed certificate",
+			httpConf: commonv1beta1.HTTPConfig{
+				TLS: commonv1beta1.TLSOptions{
+					SelfSignedCertificate: &commonv1beta1.SelfSignedCertificate{
+						SubjectAlternativeNames: []commonv1beta1.SubjectAlternativeName{
+							{
+								DNS: "elasticsearch-test.local",
+							},
+						},
+					},
+				},
+			},
+			wantSvc: func() corev1.Service {
+				svc := mkService()
+				svc.Spec.Ports[0].Name = "https"
+				return svc
+			},
+		},
+		{
+			name: "user-provided certificate",
+			httpConf: commonv1beta1.HTTPConfig{
+				TLS: commonv1beta1.TLSOptions{
+					Certificate: commonv1beta1.SecretRef{
+						SecretName: "my-cert",
+					},
+				},
+			},
+			wantSvc: func() corev1.Service {
+				svc := mkService()
+				svc.Spec.Ports[0].Name = "https"
```

Comment:
nit: would passing the name as argument help a little bit in reading the tests? `mkService("https")`

Reply:
I kept the function generic because there are a few more tests that I am planning to add in a future PR.